pages/dashboard/[guildId]/[tokenId]/configs/[configId].tsx

```diff
@@ -0,0 +1,341 @@
+    starkyModuleConfig: "",
+  });
+  const [isSaving, setIsSaving] = useState(false);
+  const [saveError, setSaveError] = useState<string | null>(null);
+  const [saveSuccess, setSaveSuccess] = useState(false);
+
+  useEffect(() => {
+    if (config) {
+      setFormData({
+        starknetNetwork: config.starknetNetwork,
+        discordRoleId: config.discordRoleId,
+        starkyModuleType: config.starkyModuleType,
+        starkyModuleConfig: JSON.stringify(config.starkyModuleConfig, null, 2),
+      });
+    }
+  }, [config]);
+
+  if (error === "Invalid or expired token.") {
+    return (
+      <RedirectMessage
+        title="Session Expired"
+        description="Your access token has expired. You'll be redirected shortly."
+        redirectTo="/"
+      />
+    );
+  }
+
+  if (error) {
+    return (
+      <div className={styles.container}>
+        <Logo />
+        <h1>Error</h1>
+        <p>{error}</p>
+        <SocialLinks />
+      </div>
+    );
+  }
+
+  const handleInputChange = (
+    e: React.ChangeEvent<
+      HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement
+    >
+  ) => {
+    const { name, value } = e.target;
+    setFormData((prev) => ({
+      ...prev,
+      [name]: value,
+    }));
+  };
+
+  const handleSubmit = async (e: FormEvent) => {
+    e.preventDefault();
+    setIsSaving(true);
+    setSaveError(null);
+    setSaveSuccess(false);
+
+    try {
+      // Parse the JSON config
+      let parsedConfig;
+      try {
+        parsedConfig = JSON.parse(formData.starkyModuleConfig);
+      } catch (err) {
+        setSaveError("Invalid JSON in module config");
+        setIsSaving(false);
+        return;
+      }
+
+      // Submit the form data
+      const response = await axios.put(
+        `/api/guilds/${guildId}/configs/${config.id}?token=${tokenId}`,
+        {
+          starknetNetwork: formData.starknetNetwork,
+          discordRoleId: formData.discordRoleId,
+          starkyModuleType: formData.starkyModuleType,
+          starkyModuleConfig: parsedConfig,
+        }
+      );
+
+      if (response.status === 200) {
+        setSaveSuccess(true);
+        // Refresh the form data with the new values
+        setFormData({
+          starknetNetwork: response.data.starknetNetwork,
+          discordRoleId: response.data.discordRoleId,
+          starkyModuleType: response.data.starkyModuleType,
+          starkyModuleConfig: JSON.stringify(
+            response.data.starkyModuleConfig,
+            null,
+            2
+          ),
+        });
+      }
+    } catch (error: any) {
+      setSaveError(
+        error.response?.data?.error || "Failed to save configuration"
+      );
+    } finally {
+      setIsSaving(false);
+    }
+  };
+
+  return (
+    <div className={styles.container}>
+      <Logo />
+      <h1>Edit Configuration</h1>
+      <Guild
+        discordServerName={discordServerName!}
+        discordServerIcon={discordServerIcon}
+      />
+
+      <BackButton />
```

Comment:
Can you modify the backButton so that it accepts a "destination" parameter (default to "/"), so that instead of redirecting to the home page it redirects to the dashboard